### PR TITLE
VACMS-16937: DUW Step 1 component

### PR DESCRIPTION
**Pull request by chriskim2311**
## Are you removing, renaming or moving a folder in this PR?
- [x] No, I'm not changing any folders (skip to Summary and delete the rest of this section)
- [ ] Yes, I'm removing, renaming or moving a folder

Did you change site-wide styles, platform utilities or other infrastructure?
- [x] No
- [ ] Yes, and I used the [proxy-rewrite steps](https://github.com/department-of-veterans-affairs/vets-website/tree/main/src/applications/proxy-rewrite#that-sounds-normal-so-whats-the-proxy-all-about) to test the injected header scenario

## Summary
PR adds subway map for Step 1 of DUW

## Related issue(s)
https://github.com/department-of-veterans-affairs/va.gov-cms/issues/16937

## Testing done
Tested locally at `/discharge-upgrade-instructions/introduction`
Mural of flows: https://app.mural.co/t/departmentofveteransaffairs9999/m/departmentofveteransaffairs9999/1696352911500/80253b1c7200212cbcbcfb940d2cf4a29fed8417?sender=ua4eca0cc168eff677bd03940

## Screenshots

<details><Summary>Flow 1</Summary>
How_To_Apply_For_A_Discharge_Upgrade___Veterans_Affairs
Screenshot 2024-08-13 at 9 10 03 AM
</details>

<details><Summary>Flow 2</Summary>
How_To_Apply_For_A_Discharge_Upgrade___Veterans_Affairs
Screenshot 2024-08-13 at 9 10 03 AM
Screenshot 2024-08-13 at 9 10 03 AM
</details>

<details><Summary>Flow 3</Summary>
How_To_Apply_For_A_Discharge_Upgrade___Veterans_Affairs
Screenshot 2024-08-13 at 9 10 03 AM
</details>

<details><Summary>Flow 4</Summary>
How_To_Apply_For_A_Discharge_Upgrade___Veterans_Affairs
How_To_Apply_For_A_Discharge_Upgrade___Veterans_Affairs
Screenshot 2024-08-13 at 9 10 03 AM
</details>

<details><Summary>Flow 5</Summary>
How_To_Apply_For_A_Discharge_Upgrade___Veterans_Affairs
How_To_Apply_For_A_Discharge_Upgrade___Veterans_Affairs
</details>

<details><Summary>Flow 6</Summary>
How_To_Apply_For_A_Discharge_Upgrade___Veterans_Affairs
How_To_Apply_For_A_Discharge_Upgrade___Veterans_Affairs
</details>

<details><Summary>Flow 7</Summary>
How_To_Apply_For_A_Discharge_Upgrade___Veterans_Affairs
How_To_Apply_For_A_Discharge_Upgrade___Veterans_Affairs

</details>

## What areas of the site does it impact?

DUW v2

### Quality Assurance & Testing

- [ ] I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [ ] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [ ] Linting warnings have been addressed
- [ ] Documentation has been updated ([link to documentation](#) \*if necessary)
- [x] Screenshot of the developed feature is added
- [ ] [Accessibility testing](https://depo-platform-documentation.scrollhelp.site/developer-docs/wcag-2-1-success-criteria-and-foundational-testing) has been performed

### Error Handling

- [x] Browser console contains no warnings or errors.
- [ ] Events are being sent to the appropriate logging solution
- [ ] Feature/bug has a monitor built into Datadog or Grafana (if applicable)

### CODE REVIEW

**Review by randimays — DISMISSED** [review]
Just finished testing. I have some observations:

**Flow 1** under Screenshots in your PR description: I expected to see more text per the Mural since these two conditions aren't met

<details><summary>Flow 1 - A</summary>
<br/>
> If Question 1 = 1 or 2 AND IF Question 2= 2001-2009 only, SHOW:
> If Q1 = 8, SHOW:

Screenshot 2024-08-15 at 8 51 31 AM

The "Download and fill out DoD Form 149" text is there, but the paragraph under it is missing.

</details>

<details><summary>Flow 1 - B</summary>
<br/>
I think this text should be there too:
Screenshot 2024-08-15 at 8 55 46 AM
</details>

<details><summary>Flow 1 - C</summary>
<br/>

This text is partially there; the "Item 8" text is missing. The "Item 10" text is there but it says "Item 17" instead of "Item 10."

Screenshot 2024-08-15 at 8 58 12 AM

Screenshot 2024-08-15 at 8 58 19 AM

</details>
<hr/>

**Flow 2** under Screenshots in your PR description: I noticed that even though you selected "Navy" as the branch of service, the board later referred to is "BCMR" instead of "BCNR." I can't reproduce this in my flow, but I thought I'd point it out in case something wonky is happening.

<details><summary>Flow 2 - A</summary>
<br/>

Given that this is all pieced-together dynamic text, I'm not sure if this is possible. But if we're able to add a line break here that'd be helpful for readability. 

Screenshot 2024-08-15 at 9 05 16 AM

</details>
<details><summary>Flow 2 - B</summary>
<br/>

The text here differs a bit from the Mural.

Screenshot 2024-08-15 at 9 07 15 AM

Screenshot 2024-08-15 at 9 07 19 AM

This flow also has "Item 17" instead of "Item 10", similar to "Flow 1 - C" above.
</details>
<hr/>

**Flow 4** under Screenshots in your PR description:

<details><summary>Flow 4 - A</summary>
<br/>

Same text discrepancy as Flow 2 - B.

</details>

<details><summary>Flow 4 - B</summary>
<br/>

Some text differences between the Mural and the review instance:
Screenshot 2024-08-15 at 9 37 20 AM
Screenshot 2024-08-15 at 9 36 58 AM

</details>

<hr/>

There were a couple flows that were not in your PR description, so I wanted to make sure I covered them:

<details><summary>If Question 1 = 1 or 2 AND IF Question 2= 2001-2009 only</summary>
<br/>

Screenshot 2024-08-15 at 9 42 11 AM

While trying to get to a flow that had it, I put in these answers:

Screenshot 2024-08-15 at 9 41 44 AM

I think the prior service / honorable discharge question is missing for this flow (and possibly others?).

Screenshot 2024-08-15 at 9 46 08 AM

The very last link in this block ("Apply for a PBDR review (opens in a new tab)") goes to a page that 404s. I don't think that's necessarily our problem to solve, but we should double check that we have the right link, and if we do, point out to CAIA that it's broken.

I also noticed the same text discrepancies as Flow 2 - B and the "Item 10" vs. "Item 17" issue.

</details>

<details><summary>If DRB & 5b!=1</summary>
<br/>

I tried a flow to get this content to appear, but it didn't show up. Might've put in the wrong answers but I'm not sure.

Screenshot 2024-08-15 at 9 53 28 AM

Screenshot 2024-08-15 at 9 52 26 AM

</details>

**Comment by eselkin** [comment]
I see this in the flow 1
Screenshot 2024-08-28 at 5 29 44 PM
Is that expected? I didn't see it on the images

**Comment by chriskim2311** [comment]
@eselkin Yes that is expected. Some changes were merged from main and accounted for Randi's comments. Once the flow/functionality PR is merged I will update the screenshots in this PR

File: src/applications/discharge-wizard/components/v2/resultsComponents/StepOne.jsx [hunk 1]
```diff
@@ -0,0 +1,232 @@
...
+      name: 'TBI',
+      type: 'condition',
+    },
+    [RESPONSES.REASON_SEXUAL_ASSAULT]: {
+      name: 'sexual assault',
+      type: 'experience',
+    },
+  };
+
+  /* eslint-enable quote-props */
```
Comment (randimays) [inline src/applications/discharge-wizard/components/v2/resultsComponents/StepOne.jsx:36]: Curious; what was this added for?

**Review by randimays — COMMENTED** [review]
Nice! One question in the comments and an observation:

The results screen looks too narrow on desktop
Screenshot 2024-09-10 at 11 08 31 AM

File: src/applications/discharge-wizard/components/v2/resultsComponents/StepOne.jsx (hunk 1 above)
Comment (chriskim2311) [reply src/applications/discharge-wizard/components/v2/resultsComponents/StepOne.jsx:36]: Will remove thanks!

**Comment by laflannery** [comment]
1. Heading structure:
   - On Flow 1, there is a blue alert. Can you update the H4 in the blue alert to be an H2? This will correct the current skipped heading error.
image
   - However, on Flow 2 there is no blue alert and so the headings within the Process lists should be H2s. Is it possible to make the Process list headings conditional based on if that alert appears before or not?
image
   - If the conditional is not possible, then all the Process list headings should be H2s
2. Download form link: I know this is how it probably is on Prod but would we be able to update the Download form link to have the Download link icon before it, to make this more consistent with the rest of VA?
3. Tagging @thejordanwood as this is Design related. On Flow 7 there are 2 alerts right next to each other. I'm not a huge fan of this but since they are already here and changing it is most likely out of scope, my only recommendation would be to add some space between the two: 
image

**Comment by chriskim2311** [comment]
@laflannery Thanks! I was able to get the heading changes in. In terms of the download link icon, which link were you referring to?

**Comment by thejordanwood** [comment]
@chriskim2311 I agree with @laflannery, I think a bit of space between the alerts would be good. Let's do 16px. Here's a mockup of what that'll look like. 

Screenshot 2024-09-10 at 4 59 38 PM

**Comment by laflannery** [comment]
@chriskim2311 Headings look great, thanks! Regarding the Download icon, I was referring to these here:

image
image

And I was thinking we could add the icon before the links like this:
image

**Comment by chriskim2311** [comment]
@laflannery The RI is back up with the latest changes for the download icon. 
@randimays Can you take another look at this as well. Thanks!

**Comment by laflannery** [comment]
Approved!

**Comment by thejordanwood** [comment]
Looks good to me!

**Review by randimays — APPROVED** [review]

**Pull request merged**